### PR TITLE
Bump GHC/Stackage in examples

### PR DESCRIPTION
**Pull request by hanshoglund**
This bumps the `examples` workspace to use https://www.stackage.org/lts-15.12.

Using the Nix shell most targets in the examples workspace build fine, but `bazel build //rts/...` hangs forever. Any pointers on how to debug would be very welcome.

```
[nix-shell] $ bazel build //rts/...
INFO: Analyzed 6 targets (0 packages loaded, 0 targets configured).
INFO: Found 6 targets...
[1 / 9] HaskellBuildLibrary //rts:add-one-hs; 3s linux-sandbox
^C
```

### CODE REVIEW

File: examples/WORKSPACE [hunk 1]
```diff
@@ -106,6 +106,6 @@ stack_snapshot(
         "text",
         "text-show",
     ],
-    snapshot = "lts-14.0",
-    vendored_packages = {"split": "@split//:split"},
+    snapshot = "lts-15.2",
+    # vendored_packages = {"split": "@split//:split"},
```
Comment (hanshoglund) [inline examples/WORKSPACE:110]: Any pointers on how to regenerate this patch would be most welcome :).

File: examples/WORKSPACE [hunk 2]
```diff
@@ -64,25 +64,25 @@ cc_library(name = "z", srcs = glob(["*.c"]), hdrs = glob(["*.h"]))
...
+# haskell_cabal_library(
+#     name = "split",
+#     version = packages["split"].version,
+#     srcs = glob(["**"]),
+#     deps = packages["split"].deps,
+#     visibility = ["//visibility:public"],
+# )
+#     """,
+#     patch_args = ["-p1"],
+#     patches = ["@rules_haskell_examples//:split.patch"],
```
Comment (hanshoglund) [inline examples/WORKSPACE:81]: Any pointers on how to regenerate this patch would be most welcome :).

**Comment by aherrmann** [comment]
This includes an update of GHC to 8.8 and is hitting the same issue as reported here https://github.com/tweag/rules_haskell/issues/1323.

File: examples/WORKSPACE (hunk 2 above)
Comment (aherrmann) [reply examples/WORKSPACE:81]: The following steps should lead to a working patch:
* Find out which version of `split` is contained in the new snapshot.
* Checkout the `split` source repository at the corresponding git revision.
* Increase the version bound on `base` in that version's Cabal file.
* Generate a patch.

**Comment by aherrmann** [comment]
#1563 updated the examples to a later GHC and Stackage snapshot version.

**Pull request closed without merging**